pkg/prometheus/promcfg.go

```diff
@@ -1413,6 +1418,27 @@ func (cg *ConfigGenerator) generateRemoteReadConfig(version semver.Version, p *v
 			cfg = append(cfg, yaml.MapItem{Key: "bearer_token_file", Value: spec.BearerTokenFile})
 		}
 
+		if spec.OAuth2 != nil && version.GTE(semver.MustParse("2.27.0")) {
```

Comment:
we might want to specify this requirement in the documentation

Reply:
Good idea, I added it to the godoc of the OAuth2 type and regenerated the docs.